### PR TITLE
Ignore `schema_migrations` and `schema_info` errors

### PR DESCRIPTION
**Pull request by wenderjean**
Related to: https://github.com/hanami/hanami/issues/1087

### Changes

- Implement method `error` from STD Lib [Logger](https://github.com/ruby/logger/blob/master/lib/logger.rb#L546) in order to suppress the lines we don't want to show in the output
- Create a simple expectation to cover the new behavior

### Rational

The `Sequel` gem yields an error log for the absence of `schema_migrations` and/or `schema_info` table right before creating it and, for that reason, we can see those errors the first time we try to prepare/migrate the database even though it doesn't produce blockers.

### Inputs

The errors we are getting for each adapter are:

#### SQLite
```
SQLite3::SQLException: no such table: schema_migrations: SELECT NULL AS 'nil' FROM `schema_migrations` LIMIT 1
SQLite3::SQLException: no such table: schema_info: SELECT NULL AS 'nil' FROM `schema_info` LIMIT 1
```

#### Postgres
```
PG::UndefinedTable: ERROR:  relation "schema_migrations" does not exist
PG::UndefinedTable: ERROR:  relation "schema_info" does not exist
```

#### MySQL
```
Mysql2::Error: Table 'bookshelf_development.schema_migrations' doesn't exist: SELECT NULL AS `nil` FROM `schema_migrations` LIMIT 1
Mysql2::Error: Table 'bookshelf_development.schema_info' doesn't exist: SELECT NULL AS `nil` FROM `schema_migrations` LIMIT 1
```

### CODE REVIEW

File: lib/hanami/model/migrator/logger.rb [hunk 1]
```diff
@@ -29,6 +39,22 @@ def _format(hash)
         def initialize(stream)
           super(nil, stream: stream, formatter: Formatter.new)
         end
+
+        # @since <version>
```
Comment (wenderjean) [inline lib/hanami/model/migrator/logger.rb:43]: I'm not sure how we usually feed this information.

Comment (adam12) [reply lib/hanami/model/migrator/logger.rb:43]: Usually `# @since x.x.x`

Comment (wenderjean) [reply lib/hanami/model/migrator/logger.rb:43]: Considering the last tag is v1.3.2 I suppose I should increment a patch version and put `since 1.3.3`?

Comment (adam12) [reply lib/hanami/model/migrator/logger.rb:43]: Actually I believe it's literally `x.x.x`, and then backfilled at release.

File: lib/hanami/model/migrator/logger.rb [hunk 2]
```diff
@@ -29,6 +39,22 @@ def _format(hash)
...
+
+          super(progname, &block)
+        end
+
+        private
+
+        # @since x.x.x
+        # @api private
+        def _ignorable(progname)
+          IGNORABLE_PATTERNS.any? { |r| progname =~ r }
```
Comment (adam12) [inline lib/hanami/model/migrator/logger.rb:56]: Minor nit.

```suggestion
          IGNORABLE_PATTERNS.any? { |r| progname.match?(r) }
```

**Review by adam12 — COMMENTED** [review]
Other than minor nit, looks good to me. I only tested the SQLite integration.

**Comment by adam12** [comment]
@wenderjean is this ready to go?

**Comment by wenderjean** [comment]
It is @adam12

**Review by adam12 — APPROVED** [review]
LGTM 👍

**Comment by adam12** [comment]
Going to merge after we get a `v1.3.4` milestone added to the repo.

**Comment by adam12** [comment]
Thanks for the patch @wenderjean !

**Pull request merged**